src/toastify-es.js

```diff
@@ -0,0 +1,437 @@
+     * @param {string} [options.text=Hi there!] - Message to be displayed in the toast
+     * @param {Element} [options.node] - Provide a node to be mounted inside the toast. node takes higher precedence over text
+     * @param {number} [options.duration=3000] - Duration for which the toast should be displayed. -1 for permanent toast
+     * @param {string} [options.selector] - CSS Selector on which the toast should be added
+     * @param {url} [options.destination] - URL to which the browser should be navigated on click of the toast
+     * @param {boolean} [options.newWindow=false] - Decides whether the destination should be opened in a new window or not
+     * @param {boolean} [options.close=false] - To show the close icon or not
+     * @param {string} [options.gravity=toastify-top] - To show the toast from top or bottom
+     * @param {string} [options.position=right] - To show the toast on left or right
+     * @param {string} [options.backgroundColor] - Sets the background color of the toast
+     * @param {url} [options.avatar] - Image/icon to be shown before text
+     * @param {string} [options.className] - Ability to provide custom class name for further customization
+     * @param {boolean} [options.stopOnFocus] - To stop timer when hovered over the toast (Only if duration is set)
+     * @param {Function} [options.callback] - Invoked when the toast is dismissed
+     * @param {Function} [options.onClick] - Invoked when the toast is clicked
+     * @param {Object} [options.offset] - Ability to add some offset to axis
+     * @private
+     */
+    _init(options) {
+  
+      // Setting defaults
+      this.options = Object.assign({
+        text: "Hi there!",
+        node: undefined,
+        duration: 3000,
+        selector: undefined,
+        callback: function() {},
+        destination: undefined,
+        newWindow: false,
+        close: false,
+        gravity: "toastify-top",
+        position: "right",
+        backgroundColor: undefined,
+        avatar: "",
+        className: "",
+        stopOnFocus: undefined,
+        onClick: undefined,
+        offset: {
+          x: 0,
+          y: 0
+        }
+      }, options);
+      this.toastElement = null;
+  
+      this.options.gravity = options.gravity === "bottom" ? "toastify-bottom" : "toastify-top"; // toast position - top or bottom
+      this.options.stopOnFocus = options.stopOnFocus === undefined ? true : options.stopOnFocus; // stop timeout on focus
+  
+    }
+  
+    /**
+     * Build the Toastify element
+     * @returns {Element}
+     * @private
+     */
+    _buildToast() {
+      // Validating if the options are defined
+      if (!this.options) {
+        throw "Toastify is not initialized";
+      }
+  
+      // Creating the DOM object
+      let divElement = document.createElement("div");
+      divElement.className = "toastify on " + this.options.className;
```

Comment:
Prefer template strings?
Maybe try [Lelab](https://github.com/lebab/lebab) for some more quick changes :)

Reply:
I'm all for template strings! I'll take a gander at Lelab too, haven't used it before.